src/main/java/me/damonkelley/tictactoe/BoardFormatter.java

```diff
@@ -0,0 +1,109 @@
+package me.damonkelley.tictactoe;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static java.util.Arrays.asList;
+
+public class BoardFormatter {
+    private Board board;
+
+    public BoardFormatter(Board board) {
+        this.board = board;
+    }
+
+    public List<List<Marker>> collect() {
+        List<List<Marker>> collection = new ArrayList<>();
+
+        collection.addAll(getRows());
+        collection.addAll(getColumns());
+        collection.addAll(getDiagonals());
+        collection.addAll(getClusters());
+
+        return collection;
+    }
+
+    private List<List<Marker>> getClusters() {
+        List<List<Marker>> clusters = new ArrayList<>();
+        if (board.getSize() == 3) return clusters;
+
+        getClusterParents().forEach(space -> clusters.add(collectCluster(space)));
+        return clusters;
+    }
+
+    private List<Marker> collectCluster(Space space) {
+        int x = space.x;
+        int y = space.y;
+        return asList(
+                board.get(space),
+                board.get(new Space(x + 1, y)),
+                board.get(new Space(x, y + 1)),
+                board.get(new Space(x + 1, y + 1)));
+    }
+
+    private List<Space> getClusterParents() {
+        return board.getSpaces().entrySet()
+                .stream()
+                .filter(e -> e.getKey().y < 3 && e.getKey().x < 3)
+                .map(Map.Entry::getKey)
+                .collect(Collectors.toList());
+    }
+
+    private List<List<Marker>> getRows() {
+        return groupSpacesBy(this::getY);
+    }
+
+    private List<List<Marker>> getColumns() {
+        return groupSpacesBy(this::getX);
+    }
+
+    private List<List<Marker>> groupSpacesBy(Function<Map.Entry<Space, Marker>, Integer> getY) {
```

Comment:
I'm having a hard time following what is going on in this method. It looks to me like there are 2-3 distinct stream operations that are going on here.